### PR TITLE
feat(etl): update location fields - EUBFR-100

### PR DESCRIPTION
**Pull request by emeryro**
Add missing fields and remove empty one

### CODE REVIEW

File: services/ingestion/etl/agri/csv/test/unit/lib/__snapshots__/transform.spec.js.snap [hunk 1]
```diff
@@ -236,12 +260,16 @@ Object {
   "project_id": undefined,
   "project_locations": Array [
     Object {
+      "address": undefined,
```
Comment (yhuard) [inline services/ingestion/etl/agri/csv/test/unit/lib/__snapshots__/transform.spec.js.snap]: Are those `undefined` expected?

Comment (emeryro) [reply services/ingestion/etl/agri/csv/test/unit/lib/__snapshots__/transform.spec.js.snap]: I don't know. I just updated the snapshots as usual.

**Review by yhuard — CHANGES_REQUESTED** [review]
The snapshot tells you that the ETL does not handle `incomplete or inconsistent data` correctly.  Furthermore, `address`, `postal_code` and `town` are not extracted correctly. The code assumes there's only one of each, even when there are different countries.

The logic for the extraction of `locationArray` has to be rewritten.

**Comment by emeryro** [comment]
ETL has been updated to fix the "undefined" fields

Also there is indeed only one address, postal_code and town in data we got. If there are multiple countries, the address fields are not filled.

**Comment by degliwe** [comment]
For fields where the expected value is a `String` provide an empty string `''` instead of  `null`
For addresses the existing logic needs to be revised but first we need to confirm with the business what are the possible cases.

**Review by degliwe — DISMISSED** [review]
See comment above

**Comment by emeryro** [comment]
AGRI ETL has been updated to pass tests.
Other ETLs haven't been updated with latest flow modifications, I suggest to do it in a different ticket while adding tests on all ETLs (this is not related to this PR).

Follow-up ticket: https://webgate.ec.europa.eu/CITnet/jira/browse/EUBFR-108

File: services/ingestion/etl/budg/xls/src/lib/transform.js [hunk 2]
```diff
@@ -30,8 +30,12 @@ export default record => {
   const locationArray = record['Participating countries']
     .split(',')
     .map(country => ({
-      country_name: null,
       country_code: country,
+      region: null,
```
Comment (yhuard) [inline services/ingestion/etl/budg/xls/src/lib/transform.js]: Those defaults should be `''`, not `null`, shouldn't they?

File: services/ingestion/etl/inforegio/json/src/lib/transform.js [hunk 3]
```diff
@@ -73,10 +73,12 @@ export default record => {
     for (let i = 0; i < countryArray.length; i += 1) {
       if (previousCountries.indexOf(countryArray[i] === -1)) {
         locationArray.push({
-          country_name: null,
           country_code: countryArray[i],
           region: null,
           nuts2: null,
+          address: null,
```
Comment (yhuard) [inline services/ingestion/etl/inforegio/json/src/lib/transform.js]: Same remark as previously

File: services/ingestion/etl/types/Project.js [hunk 4]
```diff
@@ -25,16 +26,18 @@ type GeoJSON =
...
+  address?: string,
+  postal_code?: string,
+  town?: string,
   location: GeoJSON,
 };
 
 type Timeframe = {
-  from: string | null,
-  to: string | null,
+  from: string | '',
```
Comment (yhuard) [inline services/ingestion/etl/types/Project.js]: As discussed on slack, no defaults on types :)

**Review by yhuard — CHANGES_REQUESTED** [review]
Some minor changes requested

File: services/ingestion/etl/budg/xls/src/lib/transform.js (hunk 2 above)
Comment (emeryro) [reply services/ingestion/etl/budg/xls/src/lib/transform.js]: Yes, that was the point of my comment, we could update it in follow-up when adding tests to the ETL

File: services/ingestion/etl/types/Project.js (hunk 4 above)
Comment (emeryro) [reply services/ingestion/etl/types/Project.js]: So should we remove every '| ...' in this file?

Comment (kalinchernev) [reply services/ingestion/etl/types/Project.js]: I think these should also be updated

```
  total_cost?: string | null,
  private_fund?: string | null,
  public_fund?: string | null,
  other_contrib?: string | null,
  funding_area?: string | null,
```

if they were either a string or a number, then we can have unions like `total_cost?: string | number`. At any case, it's about the type of data, not default values, which I must have incorrectly used as well...

**Comment by emeryro** [comment]
PR updated, it should be fine now

**Review by yhuard — APPROVED** [review]

**Pull request merged**